### ISSUE TITLE
Add PGE File Library support from Kevsoft's "experimental-reader" branch

### ISSUE THREAD

**Issue by Wohlstand** [WohlSoft/LunaLua#22]
Some times ago Kevsoft have made an experiment to use PGE File Library instead of built-in LVL file parser. It possibly works fine in a pure engine runtime, but it works very clunky when it's running from the SMBX Editor.

* Branch where Kevsoft made his experiment: https://github.com/WohlSoft/LunaLua/tree/experimental-reader
* ~Current mainstream branch: https://github.com/WohlSoft/LunaLua/tree/experimental-npcid-extension~ **[Branch has been removed!]**
* Current working branch: https://github.com/WohlSoft/LunaLua/tree/experimental-reader-updated

It's need to take most of Kevsoft's changes and polish the whole stuff to make it be working.

**Comment by Bluenaxela** [comment]
1) experimental-npcid-extension (now changed to a [tag](/WohlSoft/LunaLua/tree/oldbranch-experimental-npcid-extension)) has been merged to [master](/WohlSoft/LunaLua/tree/master) so that master will be the current mainstream branch.


2) [experimental-reader-updated](/WohlSoft/LunaLua/tree/experimental-reader-updated) has been created, from the combination of latest [latest master](/WohlSoft/LunaLua/tree/8b695085c2584dc366e3f0f48df3123ad80daa8c) and [experimental-reader](/WohlSoft/LunaLua/tree/experimental-reader). This resolves all merge conflicts and the merged result builds successfully. It still has a very old PGE File Library though.

**Comment by Wohlstand** [comment]
1) Okay
2) Yeah, therefore is needed to put the PGE File Library as submodule or ExternalProject of CMake build.
I'll check out the whole stuff at evening,  for this moment it's morning and 9:46 AM.

**Comment by Bluenaxela** [comment]
Agreed. I just did a merge as a first step, which makes it easier to look at the changes experimental-reader had made in the context of the up-to-date code. Good to to have a proper merge that builds correctly without extra done beyond what was strictly necessary to build. Changing the PGE File Library to a submodule or something can be the next step.

**Comment by Wohlstand** [comment]
The support for level files is almost completed, the one thing is still need is:
* -[x] On side of PGE Editor fix a crash of LunaTester that unreasonable happens after level testing launch
* -[x] Implement Lua binding for PGE-FL to dump files as lua-table trees
* -[ ] Implement NPC-*.txt replacement to use PGE-FL
* -[ ] Implement World Maps replacement to use PGE-FL

**Comment by Wohlstand** [comment]
Basic PGE File Library has been added!
